packages/next/lib/verify-partytown-setup.ts

```diff
@@ -0,0 +1,101 @@
+import { promises } from 'fs'
+import chalk from 'next/dist/compiled/chalk'
+
+import path from 'path'
+import {
+  hasNecessaryDependencies,
+  NecessaryDependencies,
+} from './has-necessary-dependencies'
+import { isYarn } from './is-yarn'
+import { fileExists } from './file-exists'
+import { FatalError } from './fatal-error'
+import { recursiveDelete } from './recursive-delete'
+import * as Log from '../build/output/log'
+
+async function missingDependencyError(dir: string) {
+  throw new FatalError(
+    chalk.bold.red(
+      "It looks like you're trying to use Partytown with next/script but do not have the required package(s) installed."
+    ) +
+      '\n\n' +
+      chalk.bold(`Please install Partytown by running:`) +
+      '\n\n' +
+      `\t${chalk.bold.cyan(
+        (await isYarn(dir))
+          ? 'yarn add @builder.io/partytown'
+          : 'npm install @builder.io/partytown'
+      )}` +
+      '\n\n' +
+      chalk.bold(
+        `If you are not trying to use Partytown, please disable the experimental ${chalk.cyan(
+          '"optimizeScripts"'
+        )} flag in next.config.js.`
+      ) +
+      '\n'
+  )
+}
+
+async function copyPartytownStaticFiles(
+  publicDir: string,
+  deps: NecessaryDependencies
+) {
+  const partytownLibDir = path.join(publicDir, '~partytown')
+  const hasPartytownLibDir = await fileExists(partytownLibDir, 'directory')
+
+  if (hasPartytownLibDir) {
+    await recursiveDelete(partytownLibDir)
+    await promises.rmdir(partytownLibDir)
+  }
+
+  const { copyLibFiles } = await Promise.resolve(
+    require(path.join(deps.resolved.get('@builder.io/partytown')!, '../utils'))
+  )
+
+  await copyLibFiles(path.join(publicDir, '~partytown'))
```

Comment:
Do we want to require this be added in the public folder? It might be better suited in `.next/static` as users can have files in their public folder committed since it's not meant to be an output location

Reply:
Thanks, that's a good point. There's no reason for these files to be committed so I just [updated it](https://github.com/vercel/next.js/pull/34244/files#diff-503b31fa14e16da2eff70176a73da5d351caaa0f714f1ad67e61590024568801R399) to copy to `.next/static/~partytown` instead.